lib/DiscoveryStreamFeed.jsm

```diff
@@ -145,25 +145,48 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     }
   }
 
+  setupFeedPromise(url, newFeeds) {
+    const feedPromise = this.getComponentFeed(url);
+
+    feedPromise.then(data => {
+      newFeeds[url] = data;
+    }).catch(/* istanbul ignore next */ error => {
+      Cu.reportError(`Error trying to load component feed ${url}: ${error}`);
+    });
+
+    return feedPromise;
+  }
+
   async loadComponentFeeds(sendUpdate) {
     const {DiscoveryStream} = this.store.getState();
+    const newFeedsPromises = [];
     const newFeeds = {};
-    if (DiscoveryStream && DiscoveryStream.layout) {
-      for (let row of DiscoveryStream.layout) {
-        if (!row || !row.components) {
-          continue;
-        }
-        for (let component of row.components) {
-          if (component && component.feed) {
-            const {url} = component.feed;
-            newFeeds[url] = await this.getComponentFeed(url);
+
+    if (!DiscoveryStream || !DiscoveryStream.layout) {
+      // istanbul ignore next
+      return;
+    }
+
+    for (let row of DiscoveryStream.layout) {
+      if (!row || !row.components) {
+        continue;
+      }
+      for (let component of row.components) {
+        if (component && component.feed) {
+          const {url} = component.feed;
+          if (!newFeeds[url]) {
+            // We initially stub this out so we don't fetch dupes,
+            // we then fill in with the proper object inside the promise.
+            newFeeds[url] = {};
+            newFeedsPromises.push(this.setupFeedPromise(url, newFeeds));
           }
         }
       }
-
-      await this.cache.set("feeds", newFeeds);
-      sendUpdate({type: at.DISCOVERY_STREAM_FEEDS_UPDATE, data: newFeeds});
     }
+
+    await Promise.all(newFeedsPromises);
```

Comment:
It's probably OK that you don't have a catch here, since all the subpromises do have catches.  Can you add a comment explaining that to the code?  It's at least possible that the m-c eslint rules will barf on this.